app/views/UserGroup/AddUsergroupModal/index.tsx

```diff
@@ -94,55 +125,131 @@ function AddUserGroupModal(props: Props) {
+        createUsergroup,
+        {
+            loading: createUsergroupPending,
         },
-        onFailure: () => {
-            if (isDefined(initialValue?.id)) {
+    ] = useMutation<UserGroupCreateMutation, UserGroupCreateMutationVariables>(
+        USER_GROUP_CREATE,
+        {
+            onCompleted: (response) => {
+                if (!response?.userGroupCreate?.ok) {
+                    return;
+                }
+
+                const responseId = response?.userGroupCreate?.result?.id;
+                if (responseId) {
+                    onSuccess(responseId);
+                }
+                const {
+                    ok,
+                    errors,
+                } = response && response.userGroupCreate;
+
+                if (errors) {
+                    const formError = transformToFormError(removeNull(errors) as ObjectError[]);
+                    setError(formError);
+                    alert.show(
+                        'Failed to create user group.',
+                        { variant: 'error' },
+                    );
+                } else if (ok) {
+                    alert.show(
+                        'Successfully created user group!',
+                        { variant: 'success' },
+                    );
+                }
+                onModalClose();
+            },
+            onError: () => {
                 alert.show(
-                    'Failed to edit user group.',
+                    'Failed to create user group.',
                     { variant: 'error' },
                 );
-            } else {
+            },
+        },
+    );
+
+    const [
+        updateUsergroup,
+        {
+            loading: updateUsergroupPending,
+        },
+    ] = useMutation<UserGroupUpdateMutation, UserGroupUpdateMutationVariables>(
+        USER_GROUP_UPDATE,
+        {
+            onCompleted: (response) => {
+                if (!response?.userGroup?.userGroupUpdate?.ok) {
+                    return;
+                }
+
+                const responseId = response?.userGroup?.userGroupUpdate?.result?.id;
+                if (responseId) {
+                    onSuccess(responseId);
+                }
+
+                const responseData = response?.userGroup?.userGroupUpdate;
+
+                const {
+                    ok,
+                    errors,
+                } = responseData;
+
+                if (errors) {
+                    const formError = transformToFormError(removeNull(errors) as ObjectError[]);
+                    setError(formError);
+                    alert.show(
+                        'Failed to update user group.',
+                        { variant: 'error' },
+                    );
+                } else if (ok) {
+                    alert.show(
+                        'Successfully updated user group!',
+                        { variant: 'success' },
+                    );
+                }
+                onModalClose();
```

Comment:
We should try to not call too many callbacks.

Also, we are not handling the case when error occurs.